DasharoPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -1067,6 +1070,168 @@ WarnIfRecoveryBoot (
   BootLogoEnableLogo ();
 }
 
+
+
+typedef struct {
+  TPM_ALG_ID AlgId;
+  CHAR16    *Name;
+} HASH_ALG_ENTRY;
+
+STATIC CONST HASH_ALG_ENTRY mHashAlgs[] = {
+  { TPM_ALG_SHA1,    L"SHA1"    },
+  { TPM_ALG_SHA256,  L"SHA256"  },
+  { TPM_ALG_SHA384,  L"SHA384"  },
+  { TPM_ALG_SHA512,  L"SHA512"  },
+  { TPM_ALG_SM3_256, L"SM3_256" },
+};
+
+STATIC
+VOID
+WarnIfSinglePCRBank (
+  VOID
+)
+{
+  EFI_STATUS     Status;
+  EFI_EVENT      Events[1];
+  BOOLEAN        CursorVisible;
+  UINTN          CurrentAttribute;
+  UINTN          Index;
+  EFI_INPUT_KEY  Key;
+  UINT32         TpmHashAlgorithmBitmap;
+  UINT32         ActivePcrBanks;
+  UINT32         RequestedActivePcrBanks;
+  UINTN          Size;
+  UINTN          i;
+  UINTN          OptionCount = 0;
+  UINTN          AvlIdx[ARRAY_SIZE(mHashAlgs)] = {0};
+  CHAR16         OptLine[81];
+  UINT32         SelectedMask;
+
+  Status = Tpm2GetCapabilitySupportedAndActivePcrs(&TpmHashAlgorithmBitmap, &ActivePcrBanks);
+  if (EFI_ERROR(Status)) {
+      return;
+  }

```

Comment:
I think placing these 4 lines after `gRT->SetVariable` is better.  This avoids bothering TPM on boots where PCR banks aren't being changed (so most of the boots) and prevents potential delays associated with that.